frontend/public/components/monitoring/targets.tsx

```diff
@@ -284,56 +297,92 @@ const Row: React.FC<RowFunctionArgs<Target>> = ({ obj }) => {
+        props: { className: tableClasses[0] },
+      },
+      {
+        id: 'monitor',
+        title: t('public~Monitor'),
+        props: { className: tableClasses[1] },
+      },
+      {
+        id: 'health',
+        title: t('public~Status'),
+        sort: 'health',
+        transforms: [sortable],
+        props: { className: tableClasses[2] },
+      },
+      {
+        id: 'namespace',
+        title: t('public~Namespace'),
+        sort: 'labels.namespace',
+        transforms: [sortable],
+        props: { className: tableClasses[3] },
+      },
+      {
+        id: 'lastScrape',
+        title: t('public~Last Scrape'),
+        sort: 'lastScrape',
+        transforms: [sortable],
+        props: { className: tableClasses[4] },
+      },
+      {
+        id: 'lastScrapeDuration',
+        title: t('public~Scrape Duration'),
+        sort: 'lastScrapeDuration',
+        transforms: [sortable],
+        props: { className: tableClasses[5] },
+      },
+    ],
+    [t],
+  );
 
-const List: React.FC<ListProps> = ({ loaded, loadError, targets }) => {
-  const { t } = useTranslation();
+  return (
+    <VirtualizedTable<Target>
+      aria-label="metrics targets"
+      columns={columns}
+      data={data}
+      loaded={loaded}
+      loadError={loadError}
+      Row={Row}
+      unfilteredData={unfilteredData}
+    />
+  );
+};
 
-  const filters = useSelector(({ k8s }: RootState) => k8s.getIn([REDUX_ID, 'filters']));
+const fuzzyCaseInsensitive = (a: string, b: string): boolean => fuzzy(_.toLower(a), _.toLower(b));
 
-  const Header = () => [
-    {
-      title: t('public~Endpoint'),
-      sortField: 'scrapeUrl',
-      transforms: [sortable],
-      props: { className: tableClasses[0] },
-    },
-    {
-      title: t('public~Monitor'),
-      props: { className: tableClasses[1] },
-    },
-    {
-      title: t('public~Status'),
-      sortField: 'health',
-      transforms: [sortable],
-      props: { className: tableClasses[2] },
-    },
-    {
-      title: t('public~Namespace'),
-      sortField: 'labels.namespace',
-      transforms: [sortable],
-      props: { className: tableClasses[3] },
-    },
-    {
-      title: t('public~Last Scrape'),
-      sortField: 'lastScrape',
-      transforms: [sortable],
-      props: { className: tableClasses[4] },
-    },
-    {
-      title: t('public~Scrape Duration'),
-      sortFunc: 'targetScrapeDuration',
-      transforms: [sortable],
-      props: { className: tableClasses[5] },
-    },
-  ];
+type ListPageProps = {
+  loaded: boolean;
+  loadError: string;
+  targets: Target[];
+};
+
+const ListPage: React.FC<ListPageProps> = ({ loaded, loadError, targets }) => {
+  const { t } = useTranslation();
+
+  const nameFilter = {
+    filter: (filter, target: Target) =>
+      fuzzyCaseInsensitive(filter.selected?.[0], target.scrapeUrl) ||
+      fuzzyCaseInsensitive(filter.selected?.[0], target.labels?.namespace),
+    items: [],
+    type: 'name',
+  };
```

Comment:
Maybe it will be better to cast this to a `RowFilter`

```suggestion
  const nameFilter: RowFilter = {
    filter: (filter, target: Target) =>
      fuzzyCaseInsensitive(filter.selected?.[0], target.scrapeUrl) ||
      fuzzyCaseInsensitive(filter.selected?.[0], target.labels?.namespace),
    items: [],
    type: 'name',
  } as RowFilter;
```

Reply:
👍